packages/suite/src/components/firmware/ReconnectDevicePrompt.tsx

```diff
@@ -142,12 +147,12 @@ export const ReconnectDevicePrompt = ({ onClose, onSuccess }: ReconnectDevicePro
 
     return (
         <Modal.Backdrop onClick={isAbortable ? onClose : undefined}>
-            {!isManualRebootRequired && !isRebootDone && (
+            {showConfirmOnDevice && (
                 <ConfirmOnDevice
                     title={<Translation id="TR_CONFIRM_ON_TREZOR" />}
-                    deviceModelInternal={deviceModelInternal}
-                    deviceUnitColor={device?.features?.unit_color}
-                    isConfirmed={!buttonEvent}
+                    deviceModelInternal={eventDevice?.features?.internal_model}
+                    deviceUnitColor={eventDevice?.features?.unit_color}
```

Comment:
Changing these because `device` is `undefined` while `deviceIsWaitingForConfirmationToConnectToHost`, so we need to use what Connect is sending in the event.